src/kbmod/masking.py

```diff
@@ -98,126 +15,70 @@ class DictionaryMasker(BitVectorMasker):
-    """Mask pixels over a given value.
-
-    Attributes
+    Parameters
     ----------
-    mask_threshold : `float`
-        The flux threshold for a pixel.
-    """
-
-    def __init__(self, mask_threshold, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.mask_threshold = mask_threshold
-
-    def apply_mask(self, stack):
-        """Apply the mask to an image stack.
-
-        Parameters
-        ----------
-        stack : `kbmod.ImageStack`
-            The stack before the masks have been applied.
+    config : `SearchConfiguration`
+        The configuration parameters
+    stack : `ImageStack`
+        The stack before the masks have been applied. Modified in-place.
 
-        Returns
-        -------
-        stack : `kbmod.ImageStack`
-            The same stack object to allow chaining.
-        """
-        stack.apply_mask_threshold(self.mask_threshold)
-        return stack
-
-
-class GrowMask(ImageMasker):
-    """Apply a mask that grows the current max out a given number of pixels.
-
-    Attributes
-    ----------
-    num_pixels : `int`
-        The number of pixels to extend the mask.
+    Returns
+    -------
+    stack : `ImageStack`
+        The stack after the masks have been applied.
     """
+    # Generate the global mask before we start modifying the individual masks.
+    if config["repeated_flag_keys"] and len(config["repeated_flag_keys"]) > 0:
+        global_flags = mask_flags_from_dict(config["mask_bits_dict"], config["repeated_flag_keys"])
+        global_binary_mask = stack.make_global_mask(global_flags, config["mask_num_images"])
+    else:
+        global_binary_mask = None
+
+    # Start by creating a binary mask out of the primary flag values. Prioritize
+    # the config's mask_bit_vector over the dictionary based version.
+    if config["mask_bit_vector"]:
+        mask_flags = config["mask_bit_vector"]
+    elif config["flag_keys"] and len(config["flag_keys"]) > 0:
+        mask_flags = mask_flags_from_dict(config["mask_bits_dict"], config["flag_keys"])
+    else:
+        mask_flags = 0
+
+    # Apply the primary mask.
+    for i in range(stack.img_count()):
+        stack.get_single_image(i).binarize_mask(mask_flags)
+
+    # If the threshold is set, mask those pixels.
+    if config["mask_threshold"]:
+        for i in range(stack.img_count()):
+            stack.get_single_image(i).union_threshold_masking(config["mask_threshold"])
+
+    # Union in the global masking if there was one.
+    if global_binary_mask is not None:
+        for i in range(stack.img_count()):
+            stack.get_single_image(i).union_masks(global_binary_mask)
+
+    # Grow the masks.
+    if config["mask_grow"] and config["mask_grow"] > 0:
+        for i in range(stack.img_count()):
+            stack.get_single_image(i).grow_mask(config["mask_grow"])
+
+    # Apply the masks to the images.
+    for i in range(stack.img_count()):
+        stack.get_single_image(i).apply_mask(0xFFFFFF)
```

Comment:
Nit: can you add a comment here to note that 0xFFFFFF is being used to apply all flags?

Reply:
Added